Feature/gui input pipelike

riedersa opened the pull request:
closes #211

riedersa committed:
added pipelike mode #211

riedersa committed:
Fixed error when starting text input and output

janschopohl commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +72,35 @@ class InputTextModel : QObject {
   int _start;
   length_t _maximumLength;
   MemoryAccess _memoryAccess;
+  int _mode;
```
There should be comments for every private member. You should add an `enum class` for mode selection (you have to register the enum if you want to use it in qml), using different Integer values is extremely error prone.

janschopohl commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +72,35 @@ class InputTextModel : QObject {
+
  private:
   QQmlContext* _context;
   int _start;
```
I don't know exaclty what this does, as there is no comment, but I suspect that using `std::size_t` instead of `int` would be better

janschopohl commented on include/ui/input-text-model.hpp:
```diff
@@ -43,6 +43,12 @@ class InputTextModel : QObject {
+   * \brief Sets a number in the memory, used for left/up/etc.
+   * \param number the number
+   */
+  Q_INVOKABLE void newNumber(int number);
```
use `std::size_t` instead of `int` here. Consider adding `using size_t = std::size_t` in this class.

janschopohl commented on source/ui/input-text-model.cpp:
```diff
@@ -25,7 +25,8 @@ InputTextModel::InputTextModel(QQmlContext* context, MemoryAccess memoryAccess)
-, _memoryAccess(memoryAccess) {
+, _memoryAccess(memoryAccess)
+, _mode(0) {
   _context->setContextProperty("inputtextMod", this);
```
This didn't change here, but `inputtextMod` is not really a great name. `inputTextModel` would be better (styleguide). (sorry for the nitpicking, but I was confused for a short moment when I saw this name in qml code)

riedersa committed:
added changes suggested from reviewer #211

I was not able to register the enum in qml. This would have needed a
constructor InputTextModel() but there is no one like this.

goldsborough commented on include/ui/input-text-model.hpp:
```diff
@@ -43,10 +46,16 @@ class InputTextModel : QObject {
    * \param text the index
    */
-  Q_INVOKABLE void setStart(unsigned int start);
+  Q_INVOKABLE void setStart(std::size_t start);
```
Since you managed to define a `length_t`, you may just want to rename the `length_t` to `size_t` and use it everywhere. Less typing.

goldsborough commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +75,54 @@ class InputTextModel : QObject {
+
+  /**
+   * \brief Gets the mode
+   * \return the length
```
😞

goldsborough commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +75,54 @@ class InputTextModel : QObject {
+  Q_INVOKABLE void setMode(int mode);
+
+  /**
+   * \brief Gets the mode
```
You don't need to use `\brief`, the first line is implicitly `brief` (up to the first dot).
Also, sentences should all end with a dot.

goldsborough commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +75,54 @@ class InputTextModel : QObject {
+
  private:
+  /**
+   * \brief the context property
```
Write this like this (on one line): `/** The context property. */`

goldsborough commented on include/ui/input-text-model.hpp:
```diff
@@ -66,17 +75,54 @@ class InputTextModel : QObject {
   void maximumLengthChanged();
+
+  /**
+   * \brief called when mode has changed
```
First letter capitalized and then a dot.

goldsborough committed:
Using size_t instead of length_t

goldsborough committed:
Changed enum class to enum for now

goldsborough committed:
Fixed stupid using directive typo in input-text-model

goldsborough commented:
I've made some changes to help out. Please fix the documentation, then it's fine.

riedersa committed:
added one comment #223

goldsborough committed:
Added size_t as a Qt Meta type

riedersa committed:
added changes suggested from reviewer #223

riedersa committed:
Merge branch 'master' into feature/gui-input-pipelike
